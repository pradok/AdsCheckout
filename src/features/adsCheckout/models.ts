import {customerModels} from '../customer';
import {productModels} from '../product';

export type Product = productModels.Product & {
  quantity: number;
  priceReduced: number;
};

export type Discount = {
  premium?: customerModels.DiscountPremium;
  classic?: customerModels.DiscountClassic;
  standout?: customerModels.DiscountStandOut;
  rulesDescription: string[];
};
