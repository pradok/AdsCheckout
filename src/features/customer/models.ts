export type DiscountPremium = {
  perAd: {
    minQty: number;
    price: number;
  };
};

export type DiscountStandOut = {
  price: number;
};

export type DiscountClassic = {
  trigger: number;
};
