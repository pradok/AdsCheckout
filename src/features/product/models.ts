export type Product = {
  id: string;
  title: string;
  price: number;
};

export interface ProductCollectionState {
  collection: Product[];
  isLoading: boolean;
  isFetched: boolean;
}
export interface ProductAction {
  type: string;
  payload?: Product[];
}
