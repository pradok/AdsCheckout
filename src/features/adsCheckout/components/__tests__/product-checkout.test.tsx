import * as React from 'react';
import {Product} from '../../models';
import {shallow} from 'enzyme';

import ProductCheckOut from '../product-checkout';

it('<Product> renders correctly', () => {
  const props: Product = {id: 'classic', price: 394.99, quantity: 379.99, priceReduced: 4, title: 'Premium Ad'};
  const component = shallow(<ProductCheckOut {...props} />);
  expect(component).toMatchSnapshot();
});
