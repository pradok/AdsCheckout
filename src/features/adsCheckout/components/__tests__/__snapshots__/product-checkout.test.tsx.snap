// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductCheckout
    id="classic"
    price={394.99}
    priceReduced={4}
    quantity={379.99}
    title="Premium Ad"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Premium Ad",
        ": $",
        4,
        " ",
        " x 379.99",
        "= ",
        "1519.96",
        "",
        "",
      ],
      "className": "classic",
    },
    "ref": null,
    "rendered": Array [
      "Premium Ad",
      ": $",
      4,
      " ",
      " x 379.99",
      "= ",
      "1519.96",
      "",
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Premium Ad",
          ": $",
          4,
          " ",
          " x 379.99",
          "= ",
          "1519.96",
          "",
          "",
        ],
        "className": "classic",
      },
      "ref": null,
      "rendered": Array [
        "Premium Ad",
        ": $",
        4,
        " ",
        " x 379.99",
        "= ",
        "1519.96",
        "",
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
